src/components/CardModal/CardModal.js

```diff
@@ -22,10 +22,10 @@ const CardModal = ({ card }) => {
   const handleCardContentOnChange = (e) => {
     setEditableContent(e.target.value);
 
-    const inputLength = e.target.value.length;
-    if (inputLength >= 20 && inputLength < 40) {
-      setWidth(e.target.value.length);
-    }
+    // The scrollHeight is the height of all the content, including that which exceeds 1 row of textarea.
+    // clientHeight is the height of the content the user can see at this point with 1 row of textarea.
+    // To dynamically set the rows, we simply divide the scrollHeight by clientHeight and roundup
+    setRows(Math.ceil(e.target.scrollHeight / e.target.clientHeight));
```

Comment:
Rows is set dynamically based on the card title. We always round up to make sure it covers hanging words or letters